tests/Util/Examples/FlagTest.cs

```diff
@@ -0,0 +1,231 @@
+        internal const string STRING_ARG1 = "stringArg1";
+        internal const string STRING_ARG2 = "stringArg2";
+        internal const string LONG_FLAG = "longFlag";
+        internal const long LONG_ARG1 = 1234;
+
+        /// <summary>
+        /// Inits this instance.
+        /// </summary>
+        [SetUp]
+        public void Init()
+        {
+        }
+
+        /// <summary>
+        /// Test for Value and Values properties for non-repeatable flags.
+        /// </summary>
+        [Test]
+        public void TestValuesForNonRepeatAbleFlag()
+        {
+            Flag<string> stringFlag = new Flag<string>($"--{STRING_FLAG}=", "", false);
+
+            // Initially, the flag is empty.
+            Assert.AreEqual(default(string), stringFlag.Value);
+
+            // After adding a value, that value comes back in both Value and Values.
+            stringFlag.Parse(STRING_ARG1);
+            Assert.AreEqual(STRING_ARG1, stringFlag.Value);
+
+            // When trying to add another value, an ArgumentException is thrown.
+            Assert.Throws(typeof(ArgumentException), 
+                delegate ()
+                {
+                    stringFlag.Parse(STRING_ARG2);
+                }
+            );
+        }
+
+        /// <summary>
+        /// Test for Value and Values properties for non-repeatable flags.
+        /// </summary>
+        [Test]
+        public void TestValuesForRepeatAbleFlag()
+        {
+            Flag<string> stringFlag = new Flag<string>($"--{STRING_REPEATED_FLAG}=", "", true);
+
+            // Initially, the flag is empty.
+            Assert.IsEmpty(stringFlag.Values);
+
+            // After adding a value, that value comes back in both Value and Values.
+            stringFlag.Parse(STRING_ARG1);
+            Assert.AreEqual(new string[] { STRING_ARG1 }, stringFlag.Values);
+            
+            // After adding another value, the value still gives the first value, but values gives
+            // the full list.
+            stringFlag.Parse(STRING_ARG2);
+            Assert.AreEqual(new string[] { STRING_ARG1, STRING_ARG2 }, stringFlag.Values);
+        }
+
+        /// <summary>
+        /// Test for Invalid operations on Value and Values properties.
+        /// </summary>
+        [Test]
+        public void TestValuesForInvalidOperation()
+        {
+            Flag<string> stringFlag = null;
+
+            // Invalid operation is thrwn if the flag is repeatable, and Value property is invoked.
+            stringFlag = new Flag<string>($"--{STRING_REPEATED_FLAG}=", "", true);
+
+            Assert.Throws(typeof(InvalidOperationException), delegate ()
+            {
+                var temp = stringFlag.Value;
+            });
+
+            // Invalid operation is thrwn if the flag is non-repeatable, and Values property is
+            // invoked.
+            stringFlag = new Flag<string>($"--{STRING_REPEATED_FLAG}=", "", false);
```

Comment:
Is this supposed to be non-repeated flag?

Reply:
The comment mentioned non-repeatable, but the actual Flag field is STRING_REPEATED_FLAG

Reply:
Done. It was testing the right thing, just that the flag name was incorrect in the context.